build.py

```diff
@@ -28,29 +32,30 @@ def install():
                       os.path.join(BIN, plugin_zip)])
     generate_package()
     p = os.listdir("dist")[0]
-    print("Installing getgauge package using pip: \n\tpip install dist/{}".format(p))
+    print(f"Installing getgauge package using pip: \n\tpip install dist/{p}")
     call([sys.executable, "-m", "pip", "install",
-          "dist/{}".format(p), "--upgrade", "--user"])
+         f"dist/{p}", "--upgrade", "--user"])
     sys.exit(exit_code)
 
 
 def create_setup_file():
-    tmpl = open("setup.tmpl", "r")
-    setup = open("setup.py", "w+")
-    v = get_version()
-    setup.write(tmpl.read().format(
-        v, "{\n\t\t':python_version == \"2.7\"': ['futures']\n\t}"))
-    setup.close()
-    tmpl.close()
+    with open("setup.tmpl", "r", encoding="utf-8") as tmpl:
+        tmpl_content = tmpl.read()
+        tmpl.close()
+    with open("setup.py", "w+", encoding="utf-8") as setup:
+        v = get_version()
+        setup.write(tmpl_content.format(
+            v, "{\n\t\t':python_version == \"2.7\"': ['futures']\n\t}"))
```

Comment:
Not really relevant for your PR, but this `python_version` declaration looks suspicious :-) Was added in 687657087ddc2df943afc88704e295e86d01ea3b so probably can go now.

Reply:
I've removed the Python 2 related part. Is this what you had in mind too?

Reply:
Ahh yeah, thanks - was more of a note for later, but thanks for doing it. Probably need to remove this line too, as this is nothing in the formatting to supply it now.

https://github.com/getgauge/gauge-python/blob/97093f64da8890548f62b17bfeec4efd36b73bae/setup.tmpl#L21